### PR TITLE
Complete invoicing plugin

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds a full invocing plugin for invoce auto generation.

TODO:
* [X] Write and enable default invocing plugin
* [X] Modify RequestInvoice to return the status so frontend knows whether invoice needs to be created async (pending) or is ready. This will have to be discussed further.
* [X] Refactor the util for PDF creation, move it to newly created plugin
* [x] Bit more refactor to entire thing
* [x] Unit tests
* [X] Create legit URL instead of mocked one in case of serving the invoice when we crerated it locally

# Impact

* [ ] New migrations
* [X] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [X] The changes are tested
* [X] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/schema.graphql [hunk 1]
```diff
@@ -1959,6 +1959,7 @@ type Invoice implements Node {
   id: ID!
   number: String
   url: String
+  status: String!
```
Comment (tomaszszymanski129) [inline saleor/graphql/schema.graphql:1962]: That's gonna get refactored to Job soon.

File: saleor/plugins/invoicing/__init__.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    chunks = []
+    for i in range(0, len(products), product_limit):
+        limit = i + product_limit
+        chunks.append(products[i:limit])
+    return chunks
+
+
+def get_product_limit_first_page(products):
+    MAX_PRODUCTS_WITH_TABLE = 3
+    MAX_PRODUCTS_WITHOUT_TABLE = 4
```
Comment (fowczarek) [inline saleor/plugins/invoicing/__init__.py:53]: Maby we should move all const outside functions?

File: saleor/plugins/invoicing/__init__.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import os
```
Comment (korycins) [inline saleor/plugins/invoicing/__init__.py:1]: Maybe separate file would be a better idea than __init__

File: tests/conftest.py [hunk 4]
```diff
@@ -180,6 +181,16 @@ def setup_vatlayer(settings):
     return settings
 
 
+@pytest.fixture
+def setup_invoicing(settings):
+    settings.PLUGINS = ["saleor.plugins.invoicing.plugin.InvoicingPlugin"]
+    data = {
+        "active": True,
```
Comment (korycins) [inline tests/conftest.py:188]: Do you need to activate it? I though it is active by default

File: saleor/plugins/invoicing/__init__.py (hunk 2 above)
Comment (tomaszszymanski129) [reply saleor/plugins/invoicing/__init__.py:53]: Not my favorite approach, in typed language we would just put `const` before and leave it inside func as it's used only here and I like it more than putting it anywhere outside - though PEP8 suggest that's the way to do it so lets go with it.

**Comment by tomaszszymanski129** [comment]
Closing in favor of https://github.com/mirumee/saleor/pull/5698, will resolve all issues there and let involved people know.

**Pull request closed without merging**